examples/kotlin/echo/src/main/kotlin/ftl/echo/Echo.kt

```diff
@@ -1,18 +1,15 @@
 package ftl.echo
 
-import ftl.builtin.Empty
-import ftl.time.time
-import xyz.block.ftl.Context
+import ftl.time.TimeClient
 import xyz.block.ftl.Export
-
-class InvalidInput(val field: String) : Exception()
+import xyz.block.ftl.Verb
 
 data class EchoRequest(val name: String?)
 data class EchoResponse(val message: String)
 
-@Throws(InvalidInput::class)
 @Export
-fun echo(context: Context, req: EchoRequest): EchoResponse {
-  val response = context.call(::time, Empty())
+@Verb
+fun echo(req: EchoRequest, time: TimeClient): EchoResponse {
```

Comment:
Looks like ctx is removed and replaced with DI?

Reply:
Yes, you directly reference the resources you need. This allows us to explicitly list this in the schema, as there is no way to get these resources without referencing them in the signature.